CMake/ConfigureBuild.cmake

```
@@ -172,11 +172,19 @@ if (UNIX)
     set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lpthread")
   endif ()
   if (OGRE_STATIC)
-    if (OGRE_CONFIG_THREADS)
-      set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lboost-thread-mt")
+    if (OGRE_CONFIG_THREADS AND OGRE_CONFIG_THREAD_PROVIDER STREQUAL "boost")
+      if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} ${Boost_THREAD_LIBRARY_DEBUG}")
+      else()
+        set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} ${Boost_THREAD_LIBRARY_RELEASE}")
+      endif()
     endif ()
     # there is no pkgconfig file for freeimage, so we need to add that lib manually
-    set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lfreeimage")
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+      set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} ${FreeImage_LIBRARY_DEBUG}")
```

```suggestion
      set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} ${FreeImage_LIBRARY}")
```

FreeImage_LIBRARY_DEBUG is not defined and the above covers both cases

---

Ah it should be `FreeImage_LIBRARY_DBG` / `FreeImage_LIBRARY_REL`: https://github.com/OGRECave/ogre/blob/master/CMake/Packages/FindFreeImage.cmake#L43-L44
I will fix this later.